### PR TITLE
Allow disabling bias for `LayerNorm`

### PR DESCRIPTION
**Pull request by janEbert**
Only relevant if `elementwise_affine=True`.

### CODE REVIEW

File: torch/nn/modules/normalization.py [hunk 1]
```diff
@@ -174,7 +176,10 @@ def __init__(self, normalized_shape: _shape_t, eps: float = 1e-5, elementwise_af
         self.elementwise_affine = elementwise_affine
         if self.elementwise_affine:
             self.weight = Parameter(torch.empty(self.normalized_shape, **factory_kwargs))
-            self.bias = Parameter(torch.empty(self.normalized_shape, **factory_kwargs))
+            if bias:
+                self.bias = Parameter(torch.empty(self.normalized_shape, **factory_kwargs))
+            else:
+                self.register_parameter('bias', None)
```
Comment (awgu) [inline torch/nn/modules/normalization.py:182]: Curious: Why register `"bias"` as `None`? This seems to set `_parameters["bias"] = None`.
https://github.com/pytorch/pytorch/blob/783a46adeeadf40b88f7450fe23daf5a7a4bb526/torch/nn/modules/module.py#L570-L571
However, we can still later set `.bias = nn.Parameter(...)` without error, so I am curious what the point of calling `register_parameter("bias", None)` is.

Comment (albanD) [reply torch/nn/modules/normalization.py:182]: This ensures that the name is reserved for a Parameter and make the Module more consistent with and without the arg passed. We do this for linear/conv as well.

**Comment by jbschlosser** [comment]
Hey @janEbert, thanks for the PR - is there an associated issue for this?

**Comment by mikaylagawarecki** [comment]
For my understanding, is this change related to the porting of RMSNorm from [this comment](https://github.com/pytorch/pytorch/issues/72643#issuecomment-1492583637)?

**Comment by janEbert** [comment]
Hey, any new opinions on this? I'd be really happy to see this merged so that the PyTorch Transformers API becomes more flexible for scaling up. :)

**Comment by mikaylagawarecki** [comment]
@janEbert Apologies for the delay, so my understanding (which perhaps you were getting at) is that

`LayerNorm` is given by 

$y = \frac{x - \mathrm{E}[x]}{ \sqrt{\mathrm{Var}[x] + \epsilon}} * \gamma + \beta$
where the $\epsilon$ is added for numerical stability

This PR is doing `LayerNorm(elementwise_affine=True, bias=False)` is

$y = \frac{x - \mathrm{E}[x]}{ \sqrt{\mathrm{Var}[x] + \epsilon}} * \gamma$



and `RMSNorm` is given by 

$y = \frac{x}{ RMS(x)} * \gamma$ (with maybe an optional $+  \beta$ [per this comment](https://github.com/pytorch/pytorch/issues/72643#issuecomment-1035594258)) 

Where 

$Var(x) = \frac{1}{n}\sum\limits_{i=1}^{n}  (x_i - E[x])^2$

and 

$RMS(x) =  \sqrt{\frac{1}{n}\sum\limits_{i=1}^{n}   (x_i)^2}$ ($\epsilon$ could also be used for numerical stability in this calculation)

so `RMSNorm` defers from `LayerNorm` on 3 counts 
1) don't subtract expectation from numerator
2) use `RMS(x)` rather than $\sqrt{Var(x) + \epsilon}$ in the denominator
3) (perhaps) don't learn an elementwise affine bias 
and this change is completely separate from `RMSNorm`

In conclusion, **we are okay with adding a `bias` argument here to disable the affine bias for consistency with other modules (e.g. `Bilinear`, `Linear`)** :)

**Review by mikaylagawarecki — APPROVED** [review]

**Comment by janEbert** [comment]
Thank you so much @mikaylagawarecki, that's an amazing summary that clears up any misunderstandings!
